tea.go

```diff
@@ -1131,8 +1132,19 @@ func (p *Program) recoverFromPanic(r any) {
 	}
 	p.cancel() // Just in case a previous shutdown has failed.
 	p.shutdown(true)
-	fmt.Printf("Caught panic:\n\n%s\n\nRestoring terminal...\n\n", r)
-	debug.PrintStack()
+	// We use "\r\n" to ensure the output is formatted even when restoring the
+	// terminal does not work or when raw mode is still active.
+	rec := strings.ReplaceAll(fmt.Sprintf("%s", r), "\n", "\r\n")
+	fmt.Fprintf(os.Stderr, "Caught panic:\r\n\r\n%s\r\n\r\nRestoring terminal...\r\n\r\n", rec)
+	stack := strings.ReplaceAll(fmt.Sprintf("%s\n", debug.Stack()), "\n", "\r\n")
+	fmt.Fprint(os.Stderr, stack)
+	f, err := os.Create(fmt.Sprintf("bubbletea-panic-%d.log", time.Now().Unix()))
+	if err == nil {
+		defer f.Close()        //nolint:errcheck
+		fmt.Fprintln(f, rec)   //nolint:errcheck
+		fmt.Fprintln(f)        //nolint:errcheck
+		fmt.Fprintln(f, stack) //nolint:errcheck
+	}
```

Comment:
Question: do we want to have this enabled by default?

An easy option to make this configurable is to make it opt-in or opt-out via an env, so it can be set by the caller or the app itself:

```go
func init() {
    os.Setenv("BUBBLETEA_PANIC_TRACE_FILE", "1")
    // or
    os.Setenv("BUBBLETEA_DISABLE_PANIC_TRACE_FILE", "1")
}
```

Reply:
I'd make this opt in by default so Bubble Tea applications don't litter directories with panic notices. This behavior could be overridden with a `ProgramOption`.

Reply:
@andreynering I've made it opt-in via the `TEA_DEBUG` environment variable.